modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py

```diff
@@ -223,6 +225,42 @@ def map_axis_partitions(
             **kwargs,
         )
 
+    @classmethod
+    def _update_partition_dimension_caches(cls, partitions: np.ndarray):
+        """
+        Build and set the length and width caches of each _physical_ partition in the given list of partitions.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The partitions for which to update length caches.
+        """
+        promises = []
```

Comment:
We may want to use a different term than promises here, since we don't really use that term anywhere else in the codebase.

Reply:
changed to `futures`